test/extended/dr/backup_restore.go

```diff
@@ -0,0 +1,289 @@
+			// the private ssh key to the recovery node.
+
+			g.By("Creating a local temporary directory")
+			tempDir, err := ioutil.TempDir("", "e2e-backup-restore")
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			// Define the ssh configuration necessary to invoke scp, which does
+			// not appear to be supported by the golang ssh client.
+			commonOpts := "-o StrictHostKeyChecking=no -o LogLevel=error -o ServerAliveInterval=30 -o ConnectionAttempts=100 -o ConnectTimeout=30"
+			authOpt := fmt.Sprintf("-i %s", os.Getenv("KUBE_SSH_KEY_PATH"))
+			bastionHost := os.Getenv("KUBE_SSH_BASTION")
+			proxyOpt := ""
+			if len(bastionHost) > 0 {
+				framework.Logf("Bastion host %s will be used to proxy scp to cluster nodes", bastionHost)
+				// The bastion host is expected to be of the form address:port
+				hostParts := strings.Split(bastionHost, ":")
+				o.Expect(len(hostParts)).To(o.Equal(2))
+				address := hostParts[0]
+				port := hostParts[1]
+				// A proxy command is required for a bastion host
+				proxyOpt = fmt.Sprintf("-o ProxyCommand='ssh -A -W %%h:%%p %s %s -p %s core@%s'", commonOpts, authOpt, port, address)
+			}
+
+			g.By(fmt.Sprintf("Copying the backup directory from backup node %q to the test host", backupNode.Name))
+			backupNodeAddress := addressForNode(backupNode)
+			o.Expect(backupNodeAddress).NotTo(o.BeEmpty())
+			copyFromBackupNodeCmd := fmt.Sprintf(`scp -v %s %s %s -r core@%s:backup %s`, commonOpts, authOpt, proxyOpt, backupNodeAddress, tempDir)
+			runCommandAndRetry(copyFromBackupNodeCmd)
+
+			g.By(fmt.Sprintf("Cleaning the backup path on recovery node %q", recoveryNode.Name))
+			sudoExecOnNodeOrFail(recoveryNode, "rm -rf /home/core/backup")
+
+			g.By(fmt.Sprintf("Copying the backup directory from the test host to recovery node %q", recoveryNode.Name))
+			recoveryNodeAddress := addressForNode(recoveryNode)
+			o.Expect(recoveryNodeAddress).NotTo(o.BeEmpty())
+			copyToRecoveryNodeCmd := fmt.Sprintf(`scp %s %s %s -r %s/backup core@%s:`, commonOpts, authOpt, proxyOpt, tempDir, recoveryNodeAddress)
+			runCommandAndRetry(copyToRecoveryNodeCmd)
+
+			// Stop etcd static pods on non-recovery masters.
+			for _, master := range masters {
+				// The restore script will stop static pods on the recovery node
+				if master.Name == recoveryNode.Name {
+					continue
+				}
+				// Recovery 4b
+				g.By(fmt.Sprintf("Stopping etcd static pod on node %q", master.Name))
+				manifest := "/etc/kubernetes/manifests/etcd-pod.yaml"
+				// Move only if present to ensure idempotent behavior during debugging.
+				sudoExecOnNodeOrFail(master, fmt.Sprintf("test -f %s && mv -f %s /tmp || true", manifest, manifest))
+
+				// Recovery 4c
+				g.By(fmt.Sprintf("Waiting for etcd static pod to exit on node %q", master.Name))
+				// Look for 'etcd ' (with trailing space) to be missing to
+				// differentiate from pods like etcd-operator.
+				sudoExecOnNodeOrFail(master, "crictl ps | grep 'etcd ' | wc -l | grep -q 0")
```

Comment:
nit: technically we are waiting for the etcd process to exit. other processes could be running in the pod.

Reply:
I'm following the documented procedure. What change do you want to see to those instructions?

Reply:
Done